### DISCUSSION TITLE
Turtle on Pyodide

### DISCUSSION THREAD

**Discussion by satyamchaudharydev** [pyodide/pyodide#4181] [Ideas]
I'm interested in creating a turtle-like graphics library in Pyodide. I've noticed that we can use 'matplot' with Pyodide, but I also have a JavaScript library called jsTurtle. It shares a similar API, but it draws on an HTML CANVAS. I'm wondering how challenging it would be to develop this. I'm enthusiastic about contributing and would appreciate some guidance on how to get started.

Here's the link to the jsTurtle library for reference: [jsTurtle Library](https://hanumanum.github.io/js-turtle/index_en.html)

**Comment by rth** [comment]
See https://github.com/pyodide/pyodide/issues/1381#issuecomment-808744357 It's certainly possible.

**Reply by rth** [reply]
Ok I realize that both links in that comment are now not working. But maybe @casatir could point to where the current turtle implementation exists. I don't know if it's based on jsTurtle or not.

**Reply by cperrod** [reply]
Hi,
I allow myself to answer in place of Romain Casati : [https://basthon.fr/ ](url)
